components/camel-chatscript/src/main/java/org/apache/camel/component/ChatScriptConsumer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component;
+
+import java.util.Date;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.impl.ScheduledPollConsumer;
+
+/**
+ * The ChatScript consumer.
+ */
+public class ChatScriptConsumer extends ScheduledPollConsumer {
+    private final ChatScriptEndpoint endpoint;
+    public ChatScriptConsumer(ChatScriptEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+    }
+  
+    @Override
+    protected int poll() throws Exception {
+        Exchange exchange = endpoint.createExchange();
+       
+        // create a message body
+        exchange.getIn().setBody( this.endpoint.getCHAT_INIT());
```

Comment:
Then `from("chatbot:xxx").to("file:lines.log")` would simply create a file with a lot of CHAT_INIT messages.
It could be that chatbot is not able to consume messages, in this case we are dealing with a write-only endpoint.
([NagiosEndpoint](https://github.com/apache/camel/blob/master/components/camel-nagios/src/main/java/org/apache/camel/component/nagios/NagiosEndpoint.java) is an example of a write-only component).

Reply:
+1 to @aldettinger
To create a polling consumer you can take care about concurrent threads which are reading the same messages. If client can do this, it's good, if not, use above suggestion

Reply:
Hi @aldettinger, chatscript component should be a write only component as you have mentioned.  Chatscript will not produce any message on its own. Will make the necessary change soon.